Mathlib/Combinatorics/SimpleGraph/Sum.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2022 Iván Renison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Iván Renison
+-/
+import Mathlib.Combinatorics.SimpleGraph.Basic
+import Mathlib.Combinatorics.SimpleGraph.Maps
+
+/-!
+# Disjoint sum of graphs
+
+This file defines the disjoint sum of graphs. The disjoint sum of `G : SimpleGraph α` and
+`H : SimpleGraph β` is a graph on `α ⊕ β` where `u` and `v` are adjacent if and only if they are
+both in `G` and adjacent in `G`, or they are both in `H` and adjacent in `H`.
+
+## Main declarations
+
+* `SimpleGraph.Sum`: The disjoint sum of graphs.
+
+## Notation
+
+* `G + H`: The disjoint sum of `G` and `H`.
+-/
+
+variable {α β γ : Type*}
+
+namespace SimpleGraph
+
+/-- Disjoint sum of `G` and `H`. -/
+@[simps!]
+protected def Sum (G : SimpleGraph α) (H : SimpleGraph β) : SimpleGraph (α ⊕ β) where
+  Adj u v := match u, v with
+    | Sum.inl u, Sum.inl v => G.Adj u v
+    | Sum.inr u, Sum.inr v => H.Adj u v
+    | _, _ => false
+  symm u v := match u, v with
+    | Sum.inl u, Sum.inl v => G.adj_symm
+    | Sum.inr u, Sum.inr v => H.adj_symm
+    | Sum.inl _, Sum.inr _ | Sum.inr _, Sum.inl _ => id
+  loopless u := by cases u <;> simp
+
+/-- Disjoint sum of `G` and `H`. -/
+infixl:60 " + " => SimpleGraph.Sum
```

Comment:
I don't think it's a good idea to overload the + operator in this way. Instead, does it work to give an instance for `HAdd (SimpleGraph α) (SimpleGraph β) (SimpleGraph (α ⊕ β))`?